### PR TITLE
phpmd fixes

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

File: Controller/Adminhtml/Rules/Edit.php [hunk 1]
```diff
@@ -72,6 +72,24 @@ public function execute()
...
+
+        $this->_view->getLayout()->getBlock('dotdigitalgroup.email.rules.edit')
+            ->setData('action', $this->getUrl('*/*/save'));
+        $this->_view->renderLayout();
+    }
+
+    /**
+     * Check rule exist
+     *
+     * @param $id
```
Comment (simon-letch) [inline Controller/Adminhtml/Rules/Edit.php]: This line and the next will break code sniffer. They need a type set and doc block needs a return void/null

File: Model/ResourceModel/Contact.php [hunk 2]
```diff
@@ -752,6 +708,72 @@ private function buildColumnData($salesOrderGrid, $quote, $salesOrder, $salesOrd
     }
 
     /**
+     * Build period column
+     *
+     * @param $salesOrder
```
Comment (simon-letch) [inline Model/ResourceModel/Contact.php]: CS issue here and next line

**Comment by adeelq** [comment]
@simon-letch changes made

File: Observer/Adminhtml/ApiValidate.php [hunk 3]
```diff
@@ -86,8 +86,9 @@ public function execute(\Magento\Framework\Event\Observer $observer)
     /**
      * Validate account
      *
-     * @param $apiUsername
-     * @param $apiPassword
+     * @param mixed $apiUsername
+     * @param mixed $apiPassword
```
Comment (cdiacon) [inline Observer/Adminhtml/ApiValidate.php]: both ^^^ should be string

Comment (adeelq) [inline Observer/Adminhtml/ApiValidate.php]: actually see line 76 and line 78. it can also be boolean

Comment (cdiacon) [inline Observer/Adminhtml/ApiValidate.php]: then this should false by default 
     private function validateAccount($apiUsername = false, $apiPassword = false)

or 

string|bool

**Pull request merged**